sh/posix.bzl

```diff
@@ -181,8 +181,9 @@ MAKE_VARIABLES = "@rules_sh//sh/posix:make_variables"
 
 def _sh_posix_toolchain_impl(ctx):
     commands = {}
+    cmds = getattr(ctx.attr, "cmds", {})
```

Comment:
Another thing that can happen now, is that a user passes a command that is not known (not an element of `_commands`) and will be silently ignored. Could you add a check for this and raise an error in that case? I think it would help avoid confusing errors for users that call `sh_posix_toolchain` directly for whatever reason.

Reply:
I've added this check in 0a01038

I tested it with such a patch in rules_nixpkgs:

```patch
index 2c3d004..a48d1b2 100644
--- a/nixpkgs/nixpkgs.bzl
+++ b/nixpkgs/nixpkgs.bzl
@@ -502,13 +502,16 @@ runCommand "bazel-nixpkgs-posix-toolchain"
     cat >>$n <<EOF
     }}
     def create_posix_toolchain():
-        sh_posix_toolchain(
-            name = "nixpkgs_sh_posix",
-            cmds = {{
+        args = {{
                 cmd: discovered[cmd]
                 for cmd in posix.commands
                 if cmd in discovered
             }}
+        args["foobar"] = "/bin/foobar"
+        args["bidule"] = "/bin/bidule"
+        sh_posix_toolchain(
+            name = "nixpkgs_sh_posix",
+            cmds = args
```

Also, I considered writing a warning message if `sh_posix_toolchain` has an argument other than `cmds`, to help users migrate from the old prototype. But they'll anyway first get hit by `cmds` being missing since we made it `mandatory`.

Reply:
> Also, I considered writing a warning message if `sh_posix_toolchain` has an argument other than `cmds`, to help users migrate from the old prototype.

Bazel itself will already throw an error in that situation, so, there's no need to repeat that check.